src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,107 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String DEFAULT_LOGIN = "login";
+    private static final String DEFAULT_PASSWORD = "password";
+    private static final int DEFAULT_AGE = 18;
+    private static RegistrationService service;
+    private User testUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+        final StorageDao storageDao = new StorageDaoImpl();
+        User registeredUser = new User();
+        registeredUser.setLogin(DEFAULT_LOGIN);
+        registeredUser.setPassword(DEFAULT_PASSWORD);
+        registeredUser.setAge(DEFAULT_AGE);
+        storageDao.add(registeredUser);
+    }
+
+    @BeforeEach
+    void setUp() {
+        testUser = new User();
+        testUser.setLogin("testLogin");
+        testUser.setPassword("testPassword");
+        testUser.setAge(19);
+    }
+
+    @AfterEach
+    void clearTheStorage() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void validLoginPasswordAge_Ok() {
+        User expected = testUser;
+        User actual = service.register(testUser);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void invalidAge_NotOk() {
+        testUser.setAge(17);
+        assertThrows(RuntimeException.class, () -> {
+            service.register(testUser);
+        });
+    }
+
+    @Test
+    void invalidPassword_NotOk() {
+        testUser.setPassword("1234");
+        assertThrows(RuntimeException.class, () -> {
+            service.register(testUser);
+        });
+    }
+
+    @Test
+    void loginAlreadyAdded_NotOk() {
+        testUser.setLogin(DEFAULT_LOGIN);
+        assertThrows(RuntimeException.class, () -> {
+            service.register(testUser);
+        });
+    }
```

Comment:
did you run your tests?
I think this will fail

Reply:
year, i ran tests, and its ok

Reply:
i should get RuntimeExeption because i already added user with DEFAULT_LOGIN upper as registeredUser, and here i try to push new user with the same login